src/app/courses/add-course/add-course.component.ts

```diff
@@ -31,18 +32,28 @@ export class AddCourseComponent implements OnInit {
   }
 
   onSave(): void {
-    const course: Course = {
-      id: this.id,
-      title: this.title,
-      creationDate: this.date,
-      duration: this.duration,
+
+    let authorsList: HttpAuthor[] = [];
+
+    this.authors.forEach(author => {
+      authorsList.push({
+        id: Math.floor(1000 + Math.random() * 9000),
+        name: author
+      })
+    })
+
+    const course: HttpCourse = {
+      id: Math.floor(1000 + Math.random() * 9000),
```

Comment:
For custom id you can you this library
[uuid4](https://www.npmjs.com/package/uuid4)

Reply:
done with uuid library